lib/Immediate/Immediate.cpp

```diff
@@ -51,10 +51,20 @@
 using namespace swift;
 using namespace swift::immediate;
 
+/// The path for Swift libraries in the OS.
+#define OS_LIBRARY_PATH "/usr/lib/swift"
+
 static void *loadRuntimeLib(StringRef runtimeLibPathWithName) {
 #if defined(_WIN32)
   return LoadLibraryA(runtimeLibPathWithName.str().c_str());
 #else
+  if (!llvm::sys::path::is_absolute(runtimeLibPathWithName)) {
+    // Try an absolute path search for Swift in the OS first.
+    llvm::SmallString<128> absolutePath(OS_LIBRARY_PATH);
+    llvm::sys::path::append(absolutePath, runtimeLibPathWithName);
+    auto result = dlopen(absolutePath.c_str(), RTLD_LAZY | RTLD_GLOBAL);
+    if (result) return result;
+  }
```

Comment:
If we want to be 100% correct here, we *still* want to search the normal `dlopen` paths first, then /usr/lib/swift, then the toolchain. Maybe it's better to modify DarwinToolChains.cpp, which just sets DYLD_LIBRARY_PATH up front.

Or maybe it doesn't matter, because there's not going to be anything in /usr/lib/swift/ that conflicts with someone's own library. Hopefully.

Reply:
I think we're OK, as /usr/lib/swift will only contain the stuff that would be bundled with the compiler.